File: flutter/lib/api/internal/event/cep_plugin_sink.dart
import '../../interfaces/digia_cep_plugin.dart';
import '../../models/cep_trigger_payload.dart';
import '../../models/digia_experience_event.dart';

/// Delivers coarse [DigiaExperienceEvent]s to the registered CEP plugin.
///
/// The CEP channel is a campaign-agnostic protocol — the plugin only
/// understands Impressed/Clicked/Dismissed. The active plugin is read lazily
/// via [_activePlugin] because a plugin may be registered (or replaced) after
/// the SDK initialises; when none is registered the delivery is a clean no-op,
/// so callers fire unconditionally (SRP: this knows only how to reach the CEP).
class CepPluginSink {
  final DigiaCEPPlugin? Function() _activePlugin;

  CepPluginSink(this._activePlugin);

  void deliver(DigiaExperienceEvent event, CEPTriggerPayload payload) {
    _activePlugin()?.notifyEvent(event, payload);
  }
}

File: android/digia-engage/src/main/java/com/digia/engage/internal/event/CepPluginSink.kt
package com.digia.engage.internal.event

import com.digia.engage.CEPTriggerPayload
import com.digia.engage.DigiaExperienceEvent
import com.digia.engage.internal.PluginRegistry

/**
 * Delivers coarse [DigiaExperienceEvent]s to the registered CEP plugin.
 *
 * The CEP channel is a campaign-agnostic protocol — the plugin only understands
 * Impressed/Clicked/Dismissed. [PluginRegistry.notifyEvent] is a clean no-op
 * when no plugin is registered, so callers fire unconditionally (SRP: this knows
 * only how to reach the CEP).
 */
internal class CepPluginSink(
    private val pluginRegistry: PluginRegistry,
) {
    fun deliver(event: DigiaExperienceEvent, payload: CEPTriggerPayload) {
        pluginRegistry.notifyEvent(event, payload)
    }
}
